checks-superstaq/checks_superstaq/init_return.py

```diff
@@ -0,0 +1,58 @@
+"""Check type type annotation of __init__ functions"""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from astroid import Const, nodes
+from pylint.checkers import BaseChecker
+
+if TYPE_CHECKING:
+    from pylint.lint import PyLinter
+
+
+class InitChecker(BaseChecker):
+    """Checker for return type annotations for class constructors (__init__).
+
+    * Check that all __init__ functions have the return type annotation '-> None'.
+    """
+
+    name = "init-return-check"
+    msgs = {
+        "W9022": (
```

Comment:
out of curiosity - where did the codes come from?

Reply:
The largest code I found was W9021 which was being used for one of the docstring pylint checks. The next available one was W9022.

Reply:
ah ok thanks. is there any risk that a future pylint update might introduce a new warning conflicting with this?

Reply:
If there is a future pylint change that leads to a new warning with the same code, pylint will throw an error whenever it is called and it will state the conflicting codes. I can check if there are conventions for what the codes should be for custom plugins so that they don't cause conflicts or if there are ways of checking for free warning codes.